engineio/src/error.rs

```diff
@@ -30,20 +30,20 @@ pub enum Error {
     InvalidUrlScheme(String),
     #[error("Error during connection via http: {0}")]
     IncompleteResponseFromReqwest(#[from] ReqwestError),
+    #[error("Error with websocket connection: {0}")]
+    AsyncWebsocketError(#[from] TungsteniteError),
     #[error("Network request returned with status code: {0}")]
     IncompleteHttp(u16),
     #[error("Got illegal handshake response: {0}")]
     InvalidHandshake(String),
     #[error("Called an action before the connection was established")]
     IllegalActionBeforeOpen(),
+    #[error("Error setting up the http request: {0}")]
+    InvalidHttpRequest(#[from] http::Error),
```

Comment:
nit: Could this also be an invalid response? Or does this type of error only cover requests?

Reply:
De-facto not, but I see your point...